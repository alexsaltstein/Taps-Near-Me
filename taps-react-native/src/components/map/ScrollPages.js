import React from 'react';
import { View, TouchableOpacity, Text, Dimensions, StyleSheet } from 'react-native';
import { COLORS } from '../../styles/COLORS';

const ScrollPages = ({ title, distance = '0.7mi', navigation }) => {
  return (
    <View style={styles.scrollPage}>
      <TouchableOpacity style={styles.beerResultContainer}
        onPress={() => navigation.navigate('Venue', { id: '1231245' })}
        activeOpacity={1}>
        <Text style={styles.beerResultCarrot}>‚ü©</Text>
        <Text style={styles.beerResultTitle}>{title}</Text>
        <Text>{distance}</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  scrollPage: {
    width: Dimensions.get('window').width,
    alignItems: 'center'
  },
  beerResultContainer: {
    backgroundColor: COLORS.white,
    width: '75%',
    borderRadius: 20,
    padding: 10
  },
  beerResultCarrot: {
    color: COLORS.lightgray,
    alignSelf: 'center',
    fontSize: 20,
    marginTop: -10,
    transform: [{ rotate: '-90deg' }]
  },
  beerResultTitle: {
    fontSize: 25,
    fontFamily: 'open-sans-semi'
  },
});

export default ScrollPages;