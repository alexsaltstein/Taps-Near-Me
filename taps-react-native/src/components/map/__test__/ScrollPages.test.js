import '../../../matchMedia.mock';
import React from 'react';
import renderer, { act } from 'react-test-renderer';
import ScrollPages from '../ScrollPages';
import { RecoilRoot } from 'recoil';
import { render, fireEvent } from '@testing-library/react-native'
import { View, Text, StyleSheet, TouchableOpacity, TextInput } from 'react-native';
import ReactDOM from 'react-dom';

const mockNavigation = {
    navigate: jest.fn(),
    addListener: jest.fn()
  }

test('renders correctly', () => {
  const tree = renderer.create(<RecoilRoot><ScrollPages navigation={mockNavigation}/></RecoilRoot>).toJSON();
  expect(tree).toMatchSnapshot();
});