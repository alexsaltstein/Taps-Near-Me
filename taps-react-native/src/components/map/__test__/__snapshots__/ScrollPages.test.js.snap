// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-width-16k0tzm"
>
  <div
    className="css-view-1dbjc4n r-backgroundColor-14lw9ot r-borderRadius-y47klf r-cursor-1loqt21 r-padding-1sp51qo r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-xaq1zp"
    data-focusable={true}
    onClick={[Function]}
    onContextMenu={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "WebkitTransitionDuration": "0s",
        "transitionDuration": "0s",
      }
    }
    tabIndex="0"
  >
    <div
      className="css-text-901oao r-alignSelf-1kihuf0 r-color-16z3b8r r-fontSize-adyw6z r-marginTop-ymp9ed r-transform-ghumyc"
      dir="auto"
      onClick={[Function]}
    >
      ‚ü©
    </div>
    <div
      className="css-text-901oao r-fontFamily-18ismyl r-fontSize-1i10wst"
      dir="auto"
      onClick={[Function]}
    />
  </div>
</div>
`;
